service/frontend/workflowHandler_test.go

```
@@ -450,6 +450,57 @@ func (s *workflowHandlerSuite) TestStartWorkflowExecution_Failed_InvalidRunTimeo
 	s.Equal(errInvalidWorkflowRunTimeoutSeconds, err)
 }
 
+func (s *workflowHandlerSuite) TestStartWorkflowExecution_EnsureNonEmptyRetryPolicyInitialized() {
+	config := s.newConfig()
+	config.RPS = dc.GetIntPropertyFn(10)
+	wh := s.getWorkflowHandler(config)
+
+	startWorkflowExecutionRequest := &workflowservice.StartWorkflowExecutionRequest{
+		Namespace:  "test-namespace",
+		WorkflowId: "workflow-id",
+		WorkflowType: &commonpb.WorkflowType{
+			Name: "workflow-type",
+		},
+		TaskQueue: &taskqueuepb.TaskQueue{
+			Name: "task-queue",
+		},
+		WorkflowExecutionTimeout: timestamp.DurationPtr(1 * time.Second),
+		WorkflowRunTimeout:       timestamp.DurationPtr(time.Duration(-1) * time.Second),
+		RetryPolicy:              &commonpb.RetryPolicy{},
+		RequestId:                uuid.New(),
+	}
+	_, err := wh.StartWorkflowExecution(context.Background(), startWorkflowExecutionRequest)
+	s.Error(err)
```

[Nit] we can ignore the error here and in the test below